Main.py

```diff
@@ -248,8 +248,47 @@ def generate_wad(wad_file: str, rom_file: str, output_file: str, channel_title:
     if not os.path.isfile(wad_file):
         raise Exception("Cannot open base WAD file.")
 
+    try:
+        with open(wad_file, 'rb') as wad_stream:
+            wad_buffer = bytearray(wad_stream.read(0xFC0))
+    except FileNotFoundError as ex:
+            raise FileNotFoundError(f'Invalid path to Base WAD: "{input_file}"')
+
+    wad_app1_sha1_usa = [
+        [0x76, 0x3D, 0x4D, 0x3D, 0x07, 0x13, 0xE4, 0xD1, 0x0E, 0x44, 0x54, 0x0C, 0xCF, 0xA3, 0x25, 0x5E, 0x19, 0xF2, 0x8A, 0xF7], # US Wad App1
+    ]
+
+    wad_app1_sha1_jpn = [
+        [0x47, 0x54, 0x6E, 0x48, 0x46, 0x7A, 0xE1, 0x4D, 0x71, 0x2B, 0x8C, 0x20, 0x7E, 0x91, 0x18, 0x21, 0x58, 0x6D, 0x10, 0x43], # JP Wad App1
+    ]
+
+    wad_app5_sha1_usa = [
+        [0x7C, 0x94, 0x77, 0x69, 0x68, 0xA7, 0xE1, 0xF5, 0xFD, 0x5D, 0xC5, 0xE2, 0xB6, 0xF8, 0x32, 0xEE, 0xF4, 0x55, 0x35, 0xA0], # US Wad App5
+    ]
+
+    wad_app5_sha1_jpn = [
+        [0xD1, 0x4D, 0xEF, 0x1E, 0xCE, 0xB0, 0x6D, 0xE2, 0x05, 0xA3, 0x53, 0xC4, 0xB5, 0x66, 0xFD, 0x55, 0x9C, 0x25, 0x4F, 0x1F], # JP Wad App5
+    ]
+
+    wad_patch_name = ""
+    wad_app1_sha1 = list(wad_buffer[0xF18:0xF2C])
+    wad_app5_sha1 = list(wad_buffer[0xFA8:0xFBC])
+
+    if wad_app1_sha1 in wad_app1_sha1_usa:
+        if wad_app5_sha1 in wad_app5_sha1_usa:
+            wad_patch_name = "ootr_usa.gzi"
+        else:
+            raise Exception('Base WAD file is not a valid OoT USA or JPN wad.')
+    elif wad_app1_sha1 in wad_app1_sha1_jpn:
+        if wad_app5_sha1 in wad_app5_sha1_jpn:
+            wad_patch_name = "ootr_jpn.gzi"
+        else:
+            raise Exception('Base WAD file is not a valid OoT USA or JPN wad.')
+    else:
+        raise Exception('Base WAD file is not a valid OoT USA or JPN wad.')
```

Comment:
I think that instead of having a bunch of branches here,you should pull out the values being checked into individual booleans e.g. 
```python
isUsaApp1 = wad_app1_sha1 in wad_app1_sha1_usa
isUsaApp5 = ...
isUsaWad = isUsaApp1 && isUsaApp5
[...]
if isUsaWad:
    wad_patch_name = ...
elif isJapanWad:
    [...]
else:
    raise Exception(...)
```

Also the exception should either be a user defined class derived from Exception, RuntimeException, or one of the other [concrete exceptions](https://docs.python.org/3/library/exceptions.html#concrete-exceptions). You should not use the raw Exception base class.

Reply:
That makes more sense, sure https://github.com/OoTRandomizer/OoT-Randomizer/pull/2153/commits/1841b2239ba43211f0b230e9375fee95d3d6cb5c

Reply:
Ah, you should probably still use the repo's standard for variable naming. I wrote them using camelcase in the suggestion out of habit from work.